x-pack/plugins/ecs_data_quality_dashboard/server/routes/results/get_results.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter, Logger } from '@kbn/core/server';
+
+import { RESULTS_ROUTE_PATH, INTERNAL_API_VERSION } from '../../../common/constants';
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+import { GetResultQuery } from '../../schemas/result';
+import type { Result, ResultDocument } from '../../schemas/result';
+import { API_DEFAULT_ERROR_MESSAGE } from '../../translations';
+import type { DataQualityDashboardRequestHandlerContext } from '../../types';
+import { createResultFromDocument } from './parser';
+import { API_RESULTS_INDEX_NOT_AVAILABLE } from './translations';
+
+const getQuery = (patterns: string) => ({
+  size: 0,
+  query: {
+    bool: { filter: [{ terms: { 'rollup.pattern': patterns.split(',') } }] },
+  },
+  aggs: {
+    latest: {
+      terms: { field: 'rollup.pattern', size: 1000000 }, // big enough to get all patterns
```

Comment:
Consider a different value for `size`, because `1000000` may be invalid if the following (from the [docs](   https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-terms-query.html#:~:text=a%20remote%20index.-,By%20default%2C%20Elasticsearch%20limits%20the%20terms%20query%20to%20a%20maximum,this%20limit%20using%20the%20index.)) is applicable:

> By default, Elasticsearch limits the terms query to a maximum of 65,536 terms. This includes terms fetched using terms lookup. You can change this limit using the [index.max_terms_count](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-max-terms-count) setting.